R/exclude-potential-pcos.R

```diff
@@ -0,0 +1,54 @@
+#' Exclude metformin purchases potentially for the treatment of polycystic ovary syndrome.
+#'
+#' Takes the output from `include_gld_purchases()` and `bef` (information on sex and date
+#' of birth) to do the exclusions.
+#' This function only performs a filtering operation so outputs the same structure and
+#' variables as the input from `include_gld_purchases()`, except the addition of a logical
+#' helper variable `no_pcos` that is used in later functions.
+#' After these exclusions are made, the output is used by `exclude_pregnancy()`.
+#'
+#' @param gld_purchases The output from `include_gld_purchases()`.
+#' @param bef The `bef` register.
+#'
+#' @return The same type as the input data, default as a [tibble::tibble()].
+#'    It also has the same columns as `include_gld_purchases()`, except for a logical
+#'    helper variable `no_pcos` that is used in later functions.
+#' @keywords internal
+#' @inherit algorithm seealso
+#'
+#' @examples
+#' exclude_potential_pcos(
+#'   gld_purchases = include_gld_purchases(register_data$lmdb),
+#'   bef = register_data$bef
+#' )
+exclude_potential_pcos <- function(gld_purchases, bef) {
+  verify_required_variables(bef, "bef")
+  criteria <- get_algorithm_logic("no_potential_pcos") |>
+    # To convert the string into an R expression
+    rlang::parse_expr()
+
+  # Use the algorithm criteria to exclude potential PCOS
+  column_names_to_lower(gld_purchases) |>
+    dplyr::inner_join(column_names_to_lower(bef), by = dplyr::join_by(pnr)) |>
+    # Ensure that columns used in filtering is the correct format
+    dplyr::mutate(
+      date = as.Date(as.character(date), format = "%Y%m%d"),
+      atc = as.character(atc),
+      indication_code = as.character(indication_code),
+      koen = as.numeric(koen),
+      foed_dato = as.Date(as.character(foed_dato), format = "%Y%m%d")
+    ) |>
```

Comment:
```suggestion
```

None of this is necessary at this stage. Those should all be determined much earlier in the full pipeline. It is also probably the reason you are getting the error messages.